flake.nix

```diff
@@ -30,7 +30,7 @@
     };
 
     devShells = forAllSystems (system: let pkgs = nixpkgs.legacyPackages.${system}; in {
-      default = pkgs.callPackage ./nix/shell.nix {};
+      default = pkgs.callPackage ./nix/shell.nix {extra-pkgs = [packages.${system}.flash];};
```

Comment:
The shell script in `flash` includes an explicit reference to the firmware in the nix store, so it isn't useful for flashing the firmware produced in the dev-shell environment.

Reply:
Agree, but this is already thinking about solving [#82](https://github.com/lilyinstarlight/zmk-nix/issues/82).

The idea is this:

With this shell, one can simply call `zmk-flash-firmware` from the shell, and everything would build and flash, just like calling `nix run .#flash`, but without using nix commands (which might have to be explicitly enabled by the user by default).

Another use-case that this could help is when one can have multiple keyboards firmware defined in the config, and then, it would be able to pass specific "flash" packages to the shell.